plugins/cython/register.py

```diff
@@ -11,5 +14,22 @@ def build_file_aliases():
     )
 
 
+def _find_gather_sources():
+    for goal in Goal.all():
+        for task_name, task_type in goal.task_items():
+            if issubclass(task_type, GatherSources):
+                yield goal.name, task_name
+
 def register_goals():
-    task(name='compile-cython', action=CompileCythonCreate).install('compile')
+    gather_sources_registrations = list(_find_gather_sources())
+    if len(gather_sources_registrations) == 0:
+        raise GoalError(
+            'Failed to find a registration for the %s task to install %s before.'
+            % (GatherSources, CompileCythonCreate)
+        )
+
+    for gather_sources_goal, gather_sources_task in gather_sources_registrations:
```

Comment:
This is obviously pretty fragile since your task knows nothing otherwise of the `GatherSources` task, just of the `PythonInterpreter` and `ResolveRequirements.REQUIREMENTS_PEX` products it needs. Its unfortunately the best that can be done today.